### DISCUSSION TITLE
rerun the docker pointing to flatnode file location

### DISCUSSION THREAD

**Discussion by uswaroop4** [mediagis/nominatim-docker#280] [General]
Hi Team,
Thank you for the image, and for consolidating a lot of things to make the installation/usability easy.

My question here is can we start the docker image with the flat file ( db data)?

Steps:
1) I have run the docker build  (using 4.0 version)
2) ran the docker for canada-latest.osm.pbf , enabling flatnode

`sudo docker run -it --rm --shm-size=2g -e THREADS=8  -e NOMINATIM_FLATNODE_FILE=/nominatim/flatnode/flatnode.file  -e POSTGRES_SHARED_BUFFERS=1GB -e POSTGRES_MAINTAINENCE_WORK_MEM=1GB -e POSTGRES_AUTOVACUUM_WORK_MEM=1GB -e POSTGRES_WORK_MEM=50MB -e POSTGRES_EFFECTIVE_CACHE_SIZE=1GB -e POSTGRES_SYNCHRONOUS_COMMIT=off -e POSTGRES_MAX_WAL_SIZE=1GB -e POSTGRES_CHECKPOINT_TIMEOUT=10min -e POSTGRES_CHECKPOINT_COMPLETITION_TARGET=0.9 -e IMPORT_STYLE=address -e PBF_PATH=/nominatim/data/canada-latest.osm.pbf -e REPLICATION_URL=https://download.geofabrik.de/north-america/canada-updates/ -v /home/ec2-user/nominatim-docker/4.0/data:/nominatim/data -v  ~/Downloads/nominatim-flatnode:/nominatim/flatnode -e NOMINATIM_PASSWORD=nominatim1234 -e IMPORT_WIKIPEDIA=false  -p 8080:8080 --name nominatim nominatim`

3) I was able to successfuly query the canda geopolygon using curl

4) I have stopped my container

5) Now I want to restart a fresh container by pointing to the flatfile , so that the download doesnt happen again, but I am unable to achieve that. Is that a possible option? If yes how can we achieve that, am i missing something here?

I have tried mounting option -v ~/Downloads/nominatim-flatnode:/nominatim/flatnode, but its running from step1 again and trying to download.

Please let me know if i am missing something here.

Regards

**Comment by leonardehrenfried** [comment]
The data and flatnode volumes are not as important as the postgres one.

The readme has a section on "Persistent Container data", which has an example command: https://github.com/mediagis/nominatim-docker/tree/master/4.0#persistent-container-data

**Reply by uswaroop4** [reply]
Thanks, Leonard, for the reply.
I have read the section and applied the same, but when I run the docker, it's again doing the import, ranking, and taking the same time as before.

I was assuming that when we point to a flatnode file, it should mount the DB storage and run the docker within no time. Is my understanding wrong? Flatnode file is only to reduce the size of import?

```
      [ec2-user@ip-172-31-90-195 nominatim-flatnode]$ pwd
      /home/ec2-user/Downloads/nominatim-flatnode
      [ec2-user@ip-172-31-90-195 nominatim-flatnode]$ 

      [ec2-user@ip-172-31-90-195 nominatim-flatnode]$ ls -lrth
      total 70G
      -rwxrwxrwx 1 ec2-user ec2-user 70G Dec 24 14:22 flatnode.file
      [ec2-user@ip-172-31-90-195 nominatim-flatnode]$ 
```    
      
```
sudo docker run -it --rm --shm-size=2g  -e PBF_PATH=/nominatim/data/canada-latest.osm.pbf -e REPLICATION_URL=https://download.geofabrik.de/north-america/canada-updates/ -v ~/Downloads/nominatim-flatnode:/nominatim-data -v /home/ec2-user/nominatim-docker/4.0/data:/nominatim/data -v nominatim-data:/var/lib/postgresql/12/main -e NOMINATIM_PASSWORD=nominatim1234 -e IMPORT_WIKIPEDIA=false  -p 8080:8080 --name nominatim nominatim
```

**Reply by leonardehrenfried** [reply]
Flatnode only speeds up the import. The real data is in postgres. If you don't have the postgres data, then you will have to do an import again - there is no way around that.

**Reply by leonardehrenfried** [reply]
And if the nominatim-data volume is empty it will of course also do the import from the beginning.

**Reply by uswaroop4** [reply]
Thanks, Leonard, for all the inputs. It actually helped me.
I am actually trying to set up multiple EC2 nominatim servers, each having its own copy of Postgres, and an ALB, to route the requests.

Please point me to any common pitfalls, or better approaches to serve huge requests, with scaling.

Regards
Swaroop
